pkg/log/zap/zap_test.go

```diff
@@ -296,3 +299,135 @@ var _ = Describe("Zap logger setup", func() {
 		})
 	})
 })
+
+var _ = Describe("Zap log level flag options setup", func() {
+	var (
+		fromFlags Options
+		fs        flag.FlagSet
+	)
+
+	BeforeEach(func() {
+		fromFlags = Options{}
+		fs = *flag.NewFlagSet(os.Args[0], flag.ExitOnError) //flags are now reset
+	})
+
+	Context("with  zap-log-level options provided", func() {
+		It("Should set info/debug zap-log-level flags.", func() {
+			args := []string{"--zap-log-level=debug"}
+			fromFlags.BindFlags(&fs)
+			if err := fs.Parse(args); err != nil {
+				Expect(err).ToNot(HaveOccurred())
+			}
+			logOut := new(bytes.Buffer)
+
+			logger := New(UseFlagOptions(&fromFlags), WriteTo(logOut))
+			logger.V(0).Info("info text")
+			logger.V(1).Info("debug 1 text")
+
+			outRaw := logOut.Bytes()
+
+			Expect(string(outRaw)).Should(ContainSubstring("info text"))
+			Expect(string(outRaw)).Should(ContainSubstring("debug 1 text"))
+
+		})
+
+		It("Should set error zap-log-level flags.", func() {
+			args := []string{"--zap-log-level=error"}
+			fromFlags.BindFlags(&fs)
+			if err := fs.Parse(args); err != nil {
+				Expect(err).ToNot(HaveOccurred())
+			}
```

Comment:
```suggestion
```suggestion
			err := fs.Parse(args)
			Expect(err).ToNot(HaveOccurred())
```
```